tests/DCPS/TransientLocalMultiInstanceTest/publisher.cpp

```diff
@@ -0,0 +1,304 @@
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) create_datareader durable failed.\n")));;
+      exit(1);
+    }
+    else {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) create_datareader durable success.\n")));;
+    }
+
+    //Now create the publisher
+    DDS::Publisher_var pub =
+      participant->create_publisher(PUBLISHER_QOS_DEFAULT,
+        DDS::PublisherListener::_nil(), OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+    if (CORBA::is_nil(pub)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) create_publisher failed.\n")));;
+      exit(1);
+    }
+
+    // Create the datawriter
+    DDS::DataWriterQos dw_qos;
+    pub->get_default_datawriter_qos(dw_qos);
+    dw_qos.durability.kind = DDS::TRANSIENT_LOCAL_DURABILITY_QOS;
+    dw_qos.reliability.kind = DDS::RELIABLE_RELIABILITY_QOS;
+    dw_qos.resource_limits.max_samples_per_instance = 1000;
+    dw_qos.history.depth = 1;
+    dw_qos.history.kind = DDS::KEEP_LAST_HISTORY_QOS;
+
+    DDS::DataWriter_var dw =
+      pub->create_datawriter(topic,
+        dw_qos,
+        0,
+        OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+    if (CORBA::is_nil(dw)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) create_datawriter failed.\n")));
+      exit(1);
+    }
+
+    MessageDataWriter_var message_dw = MessageDataWriter::_narrow(dw);
+
+    const ACE_Time_Value writer_delay(0, 100 * 1000);
+
+    Message message;
+    message.subject_id = 11111;
+    DDS::InstanceHandle_t handle = message_dw->register_instance(message);
+
+    message.from = "Comic Book Guy";
+    message.subject = "Review";
+    message.text = "Worst. Movie. Ever.";
+    message.count = 1;
+
+    Message message_instance2;
+    message_instance2.subject_id = 22222;
+    DDS::InstanceHandle_t handle_instance2 = message_dw->register_instance(message_instance2);
+
+    message_instance2.from = "Comic Book Guy - Instance 2";
+    message_instance2.subject = "Review - Instance 2";
+    message_instance2.text = "Worst. Movie. Ever. - Instance 2";
+    message_instance2.count = 2;
+
+    DDS::ReturnCode_t ret = message_dw->write(message, handle);
+
+    if (ret != DDS::RETCODE_OK)
+    {
+      ACE_ERROR((LM_ERROR,
+        ACE_TEXT("(%P|%t) ERROR: %dth write() returned %d.\n"),
+        message.count, ret));
+    }
+    else {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) wrote message : %d\n"), message.count));
+    }
+
+    ret = message_dw->write(message_instance2, handle_instance2);
+
+    if (ret != DDS::RETCODE_OK) {
+      ACE_ERROR((LM_ERROR,
+        ACE_TEXT("(%P|%t) ERROR: %dth write() returned %d.\n"),
+        message_instance2.count, ret));
+    }
+    else {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) wrote message_instance2: %d\n"), message_instance2.count));
+    }
+
+    DDS::DataWriter_var dw2 =
+      pub->create_datawriter(topic,
+        dw_qos,
+        0,
+        OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+    if (CORBA::is_nil(dw2)) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) create_datawriter failed.\n")));
```

Comment:
maybe extend the logging to tell which create_datawriter failed, there are multiple ones in this client